src/shared/telemetries/errorreporter/errorsender.go

```diff
@@ -0,0 +1,66 @@
+package errorreporter
+
+import (
+	"context"
+	"github.com/Khan/genqlient/graphql"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/otterize/intents-operator/src/shared/telemetries/basicbatch"
+	"github.com/otterize/intents-operator/src/shared/telemetries/telemetriesconfig"
+	"github.com/otterize/intents-operator/src/shared/telemetries/telemetriesgql"
+	"github.com/samber/lo"
+	"github.com/spf13/viper"
+	"net/http"
+	"time"
+)
+
+type ErrorSender struct {
+	component    telemetriesgql.Component
+	errorBatcher *basicbatch.Batcher[*telemetriesgql.Error]
+	gqlClient    graphql.Client
+	enabled      bool
+}
+
+func newGqlClient() graphql.Client {
+	apiAddress := viper.GetString(telemetriesconfig.TelemetryAPIAddressKey)
+	clientTimeout := viper.GetDuration(telemetriesconfig.TimeoutKey)
+	clientWithTimeout := &http.Client{Timeout: clientTimeout}
+	return graphql.NewClient(apiAddress, clientWithTimeout)
+}
+
+func (s *ErrorSender) SendSync(errs []*telemetriesgql.Error) error {
```

Comment:
Guess we should also check here 
```suggestion
func (s *ErrorSender) SendSync(errs []*telemetriesgql.Error) error {
	if !s.enabled {
		return nil
	}

```